### PR TITLE
Day4

### PR DESCRIPTION
**Pull request by kghite**

### CODE REVIEW

File: day4/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Chapter 14
+
+No methods to fill in
+
+# Chapter 15
+
+## 15.4
+
+Fill in the body of `rehash()`
```
Comment (thecardkid) [inline day4/README.md]: Is this all they need to do? Also pls specify which class they need to add this method to

File: day4/src/MyBetterMap.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	 */
+	protected MyLinearMap<K, V> chooseMap(Object key) {
+		int index = key==null ? 0 : Math.abs(key.hashCode()) % maps.size();
+		return maps.get(index);
+	}	
+
+	@Override
+	public boolean containsKey(Object target) {
+		// to find a key, we only have to search one map
+		// TODO: Your solution here
```
Comment (thecardkid) [inline day4/src/MyBetterMap.java]: I don't see this in the readme

File: day4/src/MyBetterMap.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+		return false;
+	}
+
+	@Override
+	public boolean containsValue(Object target) {
+		// to find a value, we have to search all map
+		// TODO: Your code here
```
Comment (thecardkid) [inline day4/src/MyBetterMap.java]: See above

File: day4/src/MyLinearMap.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	}
+
+	@Override
+	public Set<Map.Entry<K, V>> entrySet() {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public V get(Object key) {
+		// TODO: Your code here
```
Comment (thecardkid) [inline day4/src/MyLinearMap.java]: Add to readme?

File: day4/src/MyLinearMap.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+		Set<K> set = new HashSet<K>();
+		for (Entry entry: entries) {
+			set.add(entry.getKey());
+		}
+		return set;
+	}
+
+	@Override
+	public V put(K key, V value) {
+		// TODO: Your solution here
```
Comment (thecardkid) [inline day4/src/MyLinearMap.java]: This?

File: solutions/src/com/allendowney/javacs/MyBetterMap.java [hunk 6]
```diff
@@ -66,18 +66,14 @@ public void clear() {
 	@Override
```
Comment (thecardkid) [inline solutions/src/com/allendowney/javacs/MyBetterMap.java]: Not sure why changes to solutions/ are added to this commit. Can you undo these changes / create a new commit without these changes?

**Comment by davidabrahams** [comment]
It seems like there's some TODOs that aren't in the README? Can you update the README?

File: day4/src/MyBetterMap.java (hunk 2 above)
Comment (kghite) [inline day4/src/MyBetterMap.java]: Fixed

File: day4/src/MyBetterMap.java (hunk 3 above)
Comment (kghite) [inline day4/src/MyBetterMap.java]: Fixed

File: day4/src/MyLinearMap.java (hunk 4 above)
Comment (kghite) [inline day4/src/MyLinearMap.java]: Not filled in until next day assignments

File: day4/src/MyLinearMap.java (hunk 5 above)
Comment (kghite) [inline day4/src/MyLinearMap.java]: See above

File: day4/src/MyLinearMap.java (hunk 4 above)
Comment (davidabrahams) [inline day4/src/MyLinearMap.java]: Maybe remove the TODO then?
